src/OptimalControlUtilities/include/BipedalLocomotionControllers/OptimalControlUtilities/ControlProblemElements.h

```diff
@@ -0,0 +1,164 @@
+/**
+ * @file ControlProblemElements.h
+ * @authors Giulio Romualdi
+ * @copyright 2019 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#ifndef BIPEDAL_LOCOMOTION_CONTROLLERS_OPTIMAL_CONTROL_UTILITIES_CONTROL_PROBLEM_ELEMENTS_H
+#define BIPEDAL_LOCOMOTION_CONTROLLERS_OPTIMAL_CONTROL_UTILITIES_CONTROL_PROBLEM_ELEMENTS_H
+
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include <iDynTree/Model/Indices.h>
+#include <iDynTree/Core/Utils.h>
+
+#include <iDynTree/Core/MatrixDynSize.h>
+
+#include <iDynTree/Core/Rotation.h>
+#include <iDynTree/Core/VectorDynSize.h>
+#include <iDynTree/KinDynComputations.h>
+
+namespace BipedalLocomotionControllers
+{
+
+namespace OptimalControlUtilities
+{
+
+/**
+ * The struct frame defines a Frame in a ControlProblemElement
+ */
+struct Frame
+{
+    iDynTree::FrameIndex indexInModel; /**< Index of the frame in the Model */
+
+    /** Index and size of the element in the ControlProblemElement */
+    iDynTree::IndexRange indexRangeInElement{iDynTree::IndexRange::InvalidRange()};
+};
+
+/**
+ * FrameNames wraps a std::pair<std::string, std::string>. It contains the label
+ * associated to the frame in the VariableHandler end the name of the frame in the model
+ */
+class FrameNames : public std::pair<std::string, std::string>
```

Comment:
I don't see the need of inheriting here. Why not including the pair instead?

Reply:
Thanks to [this line](https://github.com/dic-iit/bipedal-locomotion-controllers/pull/3/files/54ac48cfec0288c14fa311fa4417cde4e4993d57#diff-8cebb8a61d3eb43d03d8cfd120cfd235R54-R55):
```cpp
    // use the std::pair constructors
    using std::pair<std::string, std::string>::pair;
```
The `FrameNames` inherits `std::pair` constructors. For instance, the `CentroidaAngularMomentumElement` can be construct passing a `std::initializer_list<std::pair<std::string, std::string>>`
https://github.com/dic-iit/bipedal-locomotion-controllers/blob/54ac48cfec0288c14fa311fa4417cde4e4993d57/src/OptimalControlUtilities/tests/ControlProblemTest.cpp#L459-L462